python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -101,6 +100,16 @@ def is_support_edge_feat(self):
         assert self.edge_feat_name is None, 'Edge features are not supported in the ' + \
                                             f'\"{self.__class__}\" encoder.'
 
+    def is_using_edge_feat(self):
+        """ Check if an instance of this class is using edge features.
+
+        If a GraphConvEncoder child class enables edge feature support, and is provided with
+        non-None ``edge_feat_name``, will return True. This method is for functions related
+        to trainers and inferrers, e.g., ``do_full_graph_inference()``, to determine if they
+        allow to use edge features.
+        """
+        return self.edge_feat_name is not None
```

Comment:
Because the `is_support_edge_feat()` directly do assert check, check it in this method will cause assertion error even when users do not want to use edge features.

Reply:
updated the doc string to avoid confusion.